DEFEDIT-1677 Add toggle comment shortcut

vlaaad opened the pull request.

Commit:
Add toggle comment shortcut

Commit:
Use evaluation context in toggle comment check

Commit:
Add missing type hint

markusgustavssonking reviewed (commented):
Seems to work fine but the colors are off on the line under the last commented line after toggle when editing a C file. Don't know if this is new or if it was like that already. static keyword is missing orange highlight.
![colors_wrong](https://user-images.githubusercontent.com/46451484/57291399-ad4bc800-70bf-11e9-95e6-5a9beb1607c5.PNG)

matgis-king reviewed (changes requested):
Generally works well! Found some additional issues in addition to the review comments:
* Seems like this is causing some boxed math warnings.
* If I've selected a commented-out block but the selection contains empty lines, it will add an additional comment level instead of removing the comments.
* I think we should move the **Toggle Comment** menu item above **Reindent Lines** because the **Convert Indentation** item expands a submenu and it looks a bit weird to me for some reason.

Commit:
Apply review suggestions

- remove json comment toggling/highlight
- fix boxed math warnings
- remove comment on toggle when selected line set contains comments AND
  empty lines
- add comments to glsl grammar
- change "ToggleComment "menu item order

markusgustavssonking reviewed (approved):
👍
